lib/supavisor/client_handler.ex

```diff
@@ -393,15 +393,19 @@ defmodule Supavisor.ClientHandler do
 
   ## Internal functions
 
-  @spec parse_user_info(String.t()) :: {String.t() | nil, String.t()}
-  def parse_user_info(username) do
-    case :binary.matches(username, ".") do
+  @spec parse_user_info(map) :: {String.t() | nil, String.t()}
+  def parse_user_info(%{"user" => user, "options" => %{"reference" => ref}}) do
+    {user, ref}
+  end
+
+  def parse_user_info(%{"user" => user}) do
+    case :binary.matches(user, ".") do
       [] ->
-        {username, nil}
+        {user, nil}
 
       matches ->
         {pos, _} = List.last(matches)
-        {name, "." <> external_id} = String.split_at(username, pos)
+        {name, "." <> external_id} = String.split_at(user, pos)
```

Comment:
split_at splits on length, :binary.matches  returns bytes, so if you have unicode characters in the user, this may fail. To be safe, you probably want something like this:

```elixir
{pos, 1}  = List.last(matches)
<<name::size(pos)-binary, ?., external_id::binary>> = user
```

PS: I didn't test it. :)
{pos, size} = List.last(matches)

Reply:
good call, thanks!